dataRetriever/resolvers/peerAuthenticationResolver.go

```diff
@@ -0,0 +1,218 @@
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+// maxNumOfPeerAuthenticationInResponse represents max num of peer authentication messages to send
+const maxNumOfPeerAuthenticationInResponse = 50
+
+// ArgPeerAuthenticationResolver is the argument structure used to create a new peer authentication resolver instance
+type ArgPeerAuthenticationResolver struct {
+	ArgBaseResolver
+	PeerAuthenticationPool storage.Cacher
+	DataPacker             dataRetriever.DataPacker
+}
+
+// peerAuthenticationResolver is a wrapper over Resolver that is specialized in resolving peer authentication requests
+type peerAuthenticationResolver struct {
+	*baseResolver
+	messageProcessor
+	peerAuthenticationPool storage.Cacher
+	dataPacker             dataRetriever.DataPacker
+}
+
+// NewPeerAuthenticationResolver creates a peer authentication resolver
+func NewPeerAuthenticationResolver(arg ArgPeerAuthenticationResolver) (*peerAuthenticationResolver, error) {
+	err := checkArgPeerAuthenticationResolver(arg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &peerAuthenticationResolver{
+		baseResolver: &baseResolver{
+			TopicResolverSender: arg.SenderResolver,
+		},
+		messageProcessor: messageProcessor{
+			marshalizer:      arg.Marshalizer,
+			antifloodHandler: arg.AntifloodHandler,
+			throttler:        arg.Throttler,
+			topic:            arg.SenderResolver.RequestTopic(),
+		},
+		peerAuthenticationPool: arg.PeerAuthenticationPool,
+		dataPacker:             arg.DataPacker,
+	}, nil
+}
+
+func checkArgPeerAuthenticationResolver(arg ArgPeerAuthenticationResolver) error {
+	err := checkArgBase(arg.ArgBaseResolver)
+	if err != nil {
+		return err
+	}
+	if check.IfNil(arg.PeerAuthenticationPool) {
+		return dataRetriever.ErrNilPeerAuthenticationPool
+	}
+	if check.IfNil(arg.DataPacker) {
+		return dataRetriever.ErrNilDataPacker
+	}
+	return nil
+}
+
+// RequestDataFromHash requests peer authentication data from other peers having input a public key hash
+func (res *peerAuthenticationResolver) RequestDataFromHash(hash []byte, _ uint32) error {
+	return res.SendOnRequestTopic(
+		&dataRetriever.RequestData{
+			Type:  dataRetriever.HashType,
+			Value: hash,
+		},
+		[][]byte{hash},
+	)
+}
+
+// RequestDataFromHashArray requests peer authentication data from other peers having input multiple public key hashes
+func (res *peerAuthenticationResolver) RequestDataFromHashArray(hashes [][]byte, _ uint32) error {
+	b := &batch.Batch{
+		Data: hashes,
+	}
+	buffHashes, err := res.marshalizer.Marshal(b)
+	if err != nil {
+		return err
+	}
+
+	return res.SendOnRequestTopic(
+		&dataRetriever.RequestData{
+			Type:  dataRetriever.HashArrayType,
+			Value: buffHashes,
+		},
+		hashes,
+	)
+}
+
+// RequestDataFromReferenceAndChunk requests a peer authentication chunk by specifying the reference and the chunk index
+func (res *peerAuthenticationResolver) RequestDataFromReferenceAndChunk(hash []byte, chunkIndex uint32) error {
```

Comment:
I think we can have a method here like: RequestDataFromChunk and provide only the chunk index. Hash will not be required. Also, we can define a new dataRetriever.ChunkType value so we can do the filtering in the ProcessReceivedMessage function